src/FindAnApprenticeship/SFA.DAS.FindAnApprenticeship/Application/Queries/CreateAccount/SignIntoYourOldAccount/GetSignIntoYourOldAccountQueryHandler.cs

```diff
@@ -0,0 +1,27 @@
+﻿using System;
+using System.Threading;
+using System.Threading.Tasks;
+using MediatR;
+using SFA.DAS.FindAnApprenticeship.InnerApi.LegacyApi.Requests;
+using SFA.DAS.FindAnApprenticeship.InnerApi.LegacyApi.Responses;
+using SFA.DAS.SharedOuterApi.Configuration;
+using SFA.DAS.SharedOuterApi.Interfaces;
+
+namespace SFA.DAS.FindAnApprenticeship.Application.Queries.CreateAccount.SignIntoYourOldAccount;
+
+public class GetSignIntoYourOldAccountQueryHandler(
+    IFindApprenticeshipLegacyApiClient<FindApprenticeshipLegacyApiConfiguration> legacyApiClient)
+    : IRequestHandler<GetSignIntoYourOldAccountQuery, GetSignIntoYourOldAccountQueryResult>
+{
+    public async Task<GetSignIntoYourOldAccountQueryResult> Handle(GetSignIntoYourOldAccountQuery request, CancellationToken cancellationToken)
+    {
+        var result =
+            await legacyApiClient.Get<GetLegacyValidateCredentialsApiResponse>(
+                new GetLegacyValidateCredentialsApiRequest(request.Email, request.Password));
```

Comment:
I _think_ in here we should also store a record of a unsuccessful response, we can then check this cache first and see if there is more than 5 failures to just return false.

Reply:
I think also before we call this we should call the candidate account api with the email address and make sure that it hasnt already been migrated by another user.